text/src/autogluon/text/automm/predictor.py

```diff
@@ -600,28 +628,120 @@ def _fit(
-            mixup_fn: MixupModule,
+            config: Optional[dict] = None,
+            hyperparameters: Optional[Union[str, Dict, List[str]]] = None,
+            teacher_predictor: Union[str, AutoMMPredictor] = None,
+            hpo_mode: bool = False,
+            **hpo_kwargs
     ):
+        if self._config is not None:  # continuous training
+            config = self._config
+
+        config = get_config(
+            config=config,
+            overrides=hyperparameters,
+        )
+
+        if self._df_preprocessor is None:
+            df_preprocessor = init_df_preprocessor(
+                config=config.data,
+                column_types=self._column_types,
+                label_column=self._label_column,
+                train_df_x=train_df.drop(columns=self._label_column),
+                train_df_y=train_df[self._label_column],
+            )
+        else:  # continuing training
+            df_preprocessor = self._df_preprocessor
+
+        config = select_model(
+            config=config,
+            df_preprocessor=df_preprocessor
+        )
+
+        if self._data_processors is None:
+            data_processors = init_data_processors(
+                config=config,
+                df_preprocessor=df_preprocessor,
+            )
+        else:  # continuing training
+            data_processors = self._data_processors
+
+        data_processors_count = {k: len(v) for k, v in data_processors.items()}
+        logger.debug(f"data_processors_count: {data_processors_count}")
+
+        if self._model is None:
+            model = create_model(
+                config=config,
+                num_classes=self._output_shape,
+                num_numerical_columns=len(df_preprocessor.numerical_feature_names),
+                num_categories=df_preprocessor.categorical_num_categories
+            )
+        else:  # continuing training
+            model = self._model
+            
+        pos_label = try_to_infer_pos_label(
+            data_config=config.data,
+            label_encoder=df_preprocessor.label_generator,
+            problem_type=self._problem_type,
+        )
+        validation_metric, custom_metric_func = get_metric(
+            metric_name=validation_metric_name,
+            problem_type=self._problem_type,
+            num_classes=self._output_shape,
+            pos_label=pos_label,
+        )
+            
+        mixup_active, mixup_fn = get_mixup(model_config=OmegaConf.select(config,'model'),
+                                     mixup_config=OmegaConf.select(config,'data.mixup'),
+                                     num_classes=self._output_shape,
+        )
+        if mixup_active and (config.env.per_gpu_batch_size == 1 or config.env.per_gpu_batch_size % 2 == 1):
+            warnings.warn("The mixup is done on the batch."
+                          "The per_gpu_batch_size should be >1 and even for reasonable operation",
+                          UserWarning)
```

Comment:
We need to make parenthesis formats of `get_mixup` and `warnings.warn` consistent with other code. I think this is legacy format issue of previous mixup PR. We can correct it in this PR.